ports/stm/common-hal/busio/UART.c

```diff
@@ -234,6 +245,17 @@ void common_hal_busio_uart_construct(busio_uart_obj_t *self,
     errflag = HAL_OK;
 }
 
+void common_hal_busio_uart_never_reset(busio_uart_obj_t *self) {
+    for (size_t i = 0; i < MP_ARRAY_SIZE(mcu_uart_banks); i++) {
+        if (mcu_uart_banks[i] == self->handle.Instance) {
+            never_reset_uart[i] = true;
```

Comment:
Set this false in deinit. Never reset only prevents it from bulk reset. We should be able to deinit the specific object still.

Also in `deinit` you'll want to make sure it doesn't fail when freeing the static ringbuf.

Reply:
Why would you want to deinit the object? If we do, then ``` mp_raise_ValueError(translate("No TX pin"));``` is called if we try to use it...

It doesn't fail when freeing the static ringbuf, I've checked.

Reply:
We may want to do deinit to allow the user to use the pins. Anywhere that uses it should make sure the object has been deinited first.

Reply:
Ok, that makes sense. I'll fix that!

Reply:
I've thought about this a bit, we're not actually exposing this debug_uart object to python, so they won't be able to deinit it. As this would be enabled by board, and the vcp_rx/tx pins are already connected, then it's very unlikely someone would want to use those pins. Plus this would mostly be used for debugging...in which case the user would already be compiling CPY themselves. We should probably not expose the pins to CPY either?

Reply:
I'm ok not exposing the VCP pins. You should set never_reset to false though because future uses of UART never reset will expect it to work like others. Other implementations allow you to undo "never_reset" with an explicit deinit.

Reply:
Just chiming in here with some extra context: The VCP TX/RX pins can be connected to the Morpho connector on most Nucleo boards. They aren't currently exposed in Circuitpython because I don't expose Morpho connectors through Board (since they don't even have soldered pins most of the time), and assume that if they want to use it they need to access it with Microcontroller.

Reply:
But I don't think it's appropriate to say there aren't conditions where someone might want to use all pins on the Morpho connectors. The entire St-Link apparatus can be completely snapped off, after all.